jupyter_kernel_test/__init__.py

```diff
@@ -188,11 +204,111 @@ def test_display_data(self):
             raise SkipTest
 
         for sample in self.code_display_data:
-            self.flush_channels()
-            reply, output_msgs = self.execute_helper(sample['code'])
+            with self.subTest(code=sample['code']):
+                self.flush_channels()
+                reply, output_msgs = self.execute_helper(sample['code'])
+
+                self.assertEqual(reply['content']['status'], 'ok')
+
+                self.assertGreaterEqual(len(output_msgs), 1)
+                self.assertEqual(output_msgs[0]['msg_type'], 'display_data')
+                self.assertIn(sample['mime'], output_msgs[0]['content']['data'])
+
+    def test_silent(self):
+        self.flush_channels()
+
+        reply, output_msgs = self.execute_helper("", silent=True)
```

Comment:
Ditto for silent, there may well be kernels that don't implement it. Also, we should probably reuse the hello world sample, because:

- You might get bizarre behaviour when executing an empty string. I think our frontends might even short circuit that. We should probably decide and document it.
- If you don't hit corner cases, it's not producing any output anyway, so you're not really testing the silent flag.

So far, the design of jkt is that kernels opt in to each test by specifying an attribute on their subclass. That's up for discussion, but I'm inclined to keep the principle - it means we can use the latest version of jkt to test IRkernel, for instance, without worrying that new tests will suddenly appear and start breaking things.